code/modules/vehicles/tank.dm

```diff
@@ -0,0 +1,388 @@
+/*
+tankS BY KMC
+THIS IS LIKE REGULAR CM tank BUT IT'S LESS SHIT
+WHOEVER MADE CM TANKS: YOU ARE A BAD CODER!!!!!
+*/
+
+//generic component stuff
+
+/obj/item/tank_weapon //These are really just proof of concept weapons. You'll probably want to adapt them to use the hardpoint crap that CM has.
+	name = "TGS 4 main tank cannon"
+	desc = "A gun that works about 50% of the time, but at least it's open source! It fires tank shells."
+	icon = 'icons/obj/hardpoint_modules.dmi'
+	icon_state = "glauncher"
+	var/obj/item/ammo_magazine/ammo
+	var/obj/vehicle/tank/owner
+	var/list/fire_sounds = list('sound/weapons/tank_cannon_fire1.ogg', 'sound/weapons/tank_cannon_fire2.ogg')
+	var/default_ammo = /obj/item/ammo_magazine/tank/ltb_cannon //What do we start with?
+	var/list/accepted_ammo = list(/obj/item/ammo_magazine/tank/tank_glauncher) //Alternative ammo types that we'll accept. The default ammo is in this by default
+	var/fire_delay
+	var/cooldown = 60 //6 second weapons cooldown
+	var/lastfired = 0 //When were we last shot?
+
+/obj/item/tank_weapon/Initialize()
+	. = ..()
+	ammo = new default_ammo(src)
+	accepted_ammo += default_ammo
+
+/obj/item/tank_weapon/secondary_weapon
+	name = "TGS 3 XENOCRUSHER tank machine gun"
+	desc = "A much better gun that shits out bullets at ridiculous speeds, don't get in its way!"
+	icon = 'icons/obj/hardpoint_modules.dmi'
+	icon_state = "m56_cupola"
+	default_ammo = /obj/item/ammo_magazine/tank/ltaaap_minigun
+	fire_sounds = list('sound/weapons/tank_minigun_loop.ogg')
+	cooldown = 2 //Minimal cooldown
+	accepted_ammo = list(/obj/item/ammo_magazine/tank/towlauncher,/obj/item/ammo_magazine/tank/m56_cupola,/obj/item/ammo_magazine/tank/flamer,/obj/item/ammo_magazine/tank/tank_slauncher)
+
+/obj/item/tank_weapon/proc/fire(atom/T,mob/user)
+	if(!can_fire(T))
+		return FALSE
+	if(ammo.current_rounds <= 0)
```

Comment:
Check for rounds after the cooldown so we can give it a click sound for feedback without spam.